worker/caasmodelconfigmanager/worker.go

```diff
@@ -107,13 +128,60 @@ func (w *manager) Wait() error {
 	return w.catacomb.Wait()
 }
 
-func (w *manager) loop() error {
+func (w *manager) loop() (err error) {
 	controllerConfig, err := w.config.Facade.ControllerConfig()
 	if err != nil {
 		return errors.Trace(err)
 	}
-	if err := w.config.Broker.EnsureImageRepoSecret(controllerConfig.CAASImageRepo()); err != nil {
+	repoDetails := controllerConfig.CAASImageRepo()
+	if !repoDetails.IsPrivate() {
+		// No ops for public registry config.
+		return nil
+	}
+	w.reg, err = w.registryFunc(repoDetails)
+	if err != nil {
 		return errors.Trace(err)
 	}
-	return nil
+	err = w.reg.Ping()
+	if utils.IsPublicAPINotAvailableError(err) {
+		// No ops for public registry config.
+		// This should never happen because we have done this check above.
+		return nil
+	}
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if err := w.ensureImageRepoSecret(true); err != nil {
+		return errors.Trace(err)
+	}
+	for {
+		select {
+		case <-w.catacomb.Dying():
+			return w.catacomb.ErrDying()
+		case <-w.getNextTick():
+			if err := w.ensureImageRepoSecret(false); err != nil {
+				return errors.Trace(err)
+			}
+		}
+	}
+}
+
+func (w *manager) getNextTick() <-chan time.Time {
+	if w.nextTickDuration != nil {
+		return w.clock.After(*w.nextTickDuration)
+	}
+	return w.clock.After(3 * time.Second)
```

Comment:
This will cause a memory leak unless Reset() is called.

For loops should use time.NewTicker()
Set up the ticker outside the for loop

Is 3s too short as a default? Maybe 30s or something?

Reply:
Yes, you are right! fixed.
Thanks